tests/test_package_specifier.py

```diff
@@ -3,14 +3,52 @@
 import pytest  # type: ignore
 
 from pipx.package_specifier import (
+    fix_package_name,
     parse_specifier_for_install,
     parse_specifier_for_metadata,
+    valid_pypi_name,
 )
 from pipx.util import PipxError
 
 TEST_DATA_PATH = "./testdata/test_package_specifier"
 
 
+@pytest.mark.parametrize(
+    "package_spec_in,package_name_out",
+    [
+        ("Black", "black"),
+        ("https://github.com/ambv/black/archive/18.9b0.zip", None),
+        ("black @ https://github.com/ambv/black/archive/18.9b0.zip", None),
+    ],
+)
+def test_valid_pypi_name(package_spec_in, package_name_out):
+    assert valid_pypi_name(package_spec_in) == package_name_out
+
+
+@pytest.mark.parametrize(
+    "package_spec_in,package_name,package_spec_out",
+    [
+        (
+            "https://github.com/ambv/black/archive/18.9b0.zip",
+            "black",
+            "https://github.com/ambv/black/archive/18.9b0.zip",
+        ),
+        (
+            "nox@https://github.com/ambv/black/archive/18.9b0.zip",
+            "black",
+            "black@ https://github.com/ambv/black/archive/18.9b0.zip",
+        ),
+        (
+            "nox[extra]@https://github.com/ambv/black/archive/18.9b0.zip",
+            "black",
+            "black[extra]@ https://github.com/ambv/black/archive/18.9b0.zip",
```

Comment:
These moved under the psf org now, no longer under ambv.

Reply:
For these tests they are just made up urls, so it doesn't really matter if they exist, only that they are the proper syntax.

I don't think black has an extra called `extra`, I just made that up.

I wanted to find a good package with an extra I could actually try and install.